src/ctapipe/tools/stats_calculation.py

```diff
@@ -0,0 +1,179 @@
+"""
+Perform statistics calculation from DL1a image data
+"""
+
+import pathlib
+
+import numpy as np
+from astropy.table import vstack
+
+from ctapipe.core import Tool
+from ctapipe.core.tool import ToolConfigurationError
+from ctapipe.core.traits import (
+    Bool,
+    CaselessStrEnum,
+    CInt,
+    Path,
+    Set,
+    Unicode,
+    classes_with_traits,
+)
+from ctapipe.instrument import SubarrayDescription
+from ctapipe.io import write_table
+from ctapipe.io.tableloader import TableLoader
+from ctapipe.monitoring.calculator import PixelStatisticsCalculator
+
+
+class StatisticsCalculatorTool(Tool):
+    """
+    Perform statistics calculation for DL1a image data
+    """
+
+    name = "StatisticsCalculatorTool"
+    description = "Perform statistics calculation for DL1a image data"
+
+    examples = """
+    To calculate statistics of DL1a image data files:
+
+    > ctapipe-stats-calculation --input_url input.dl1.h5 --output_path /path/monitoring.h5 --overwrite
+
+    """
+
+    input_url = Path(
+        help="Input CTA HDF5 files including DL1a image data",
+        allow_none=True,
+        exists=True,
+        directory_ok=False,
+        file_ok=True,
+    ).tag(config=True)
+
+    allowed_tels = Set(
+        trait=CInt(),
+        default_value=None,
+        allow_none=True,
+        help=(
+            "List of allowed tel_ids, others will be ignored. "
+            "If None, all telescopes in the input stream will be included."
+        ),
+    ).tag(config=True)
+
+    dl1a_column_name = CaselessStrEnum(
```

Comment:
Is DL1a/b is "official"? Also, I'd perhaps use generic `input_column_name` similar to the output one.

Reply:
no, in ctapipe we use `DL1_IMAGES` and `DL1_PARAMETERS` to distinguish between things that are per-pixel vs. single quantities per event.

https://ctapipe.readthedocs.io/en/latest/api/ctapipe.io.DataLevel.html

Reply:
I'd also not make this an enum. In the generic tool, users should be able to chose any column that has compatible shape.

Reply:
This could also be a list of columns, to compute on multiple at the same time.

Reply:
Ok, I changed the column name and also polish the references to DL1a data by using pixel-wise image data which is more descriptive. `ToolConfigurationError` is raised once the column is not found. Having list of columns seems a little bit of an overkill here, which would just make the code more complex. Maybe the aggregation config could be shared between the columns, but especially the outlier detection will be different between the columns.

Reply:
> ToolConfigurationError is raised once the column is not found. Having list of columns seems a little bit of an overkill here, which would just make the code more complex. Maybe the aggregation config could be shared between the columns, but especially the outlier detection will be different between the columns.

I think the case where you only want to know about a single column is quite rare, you are usually interested in multiple. So having to read all data again to compute metrics on a new column seems very limiting and a loop over columns shouldn't make the code much more complex.